framework/src/st_things/st_things_request_handler.c

```diff
@@ -433,6 +419,19 @@ static bool get_supported_properties(const char *res_type, const char *res_uri,
 	}
 
 	return true;
+
+EXIT_WITH_ERROR:
+	res_types = NULL;
+	things_free(prop_count);
```

Comment:
Memeory free should be like this: if (prop_count) things_free(prop_count);
Otherwise, you need to modify things_free definition to check NULL of a variable.

Reply:
@davidfather  
If the pointer is null which will be free then no action occurs.

"The free function causes the space pointed to by ptr to be deallocated, that is, made available for further allocation. If ptr is a null pointer, no action occurs. Otherwise, if the argument does not match a pointer earlier returned by the calloc, malloc, or Note that this need not be the same as the representation of floating-point zero or a null pointer constant. realloc function, or if the space has been deallocated by a call to free or realloc, the behavior is undefined." - http://www.open-std.org/JTC1/SC22/wg14/www/docs/n1124.pdf